tests/unit/shared/config/ConfigTest.php

```diff
@@ -10,6 +10,12 @@
  * @covers \PharIo\Phive\Config
  */
 class ConfigTest extends TestCase {
+    protected function tearDown(): void {
+        // Unset environmet after each test
+        putenv('PHIVE_HOME');
```

Comment:
If you wouldn't have the implementation rely on global state, this wouldn't be required here.

Reply:
So within unit tests there shouldn't be any testing with real environment variables?

Reply:
Yes and no:

- When testing `Config`, a clear __no__ as it should request everything via `Environment` - which can get mocked

- When testing `Environment`, for a unit test you can instantiate `Environment` with a controlled data set (e.g. pass in a custom array to its constructor). This will allow testing of all aspects.

- When testing the factory method `fromGlobalState`, we would need actual environment variables. As that piece of code interacts with and depends on global state, this is - formally speaking - and in my opinion not a Unit but an integration test.